packages/common/src/store/pages/chat/sagas.ts

```diff
@@ -436,6 +438,75 @@ function* doCreateChat(action: ReturnType<typeof createChat>) {
   }
 }
 
+function* doCreateChatBlast(action: ReturnType<typeof createChatBlast>) {
+  const {
+    audience,
+    contentId,
+    contentType,
+    presetMessage,
+    replaceNavigation,
+    skipNavigation
+  } = action.payload
+
+  const { track, make } = yield* getContext('analytics')
+  try {
+    const currentUserId = yield* select(getUserId)
+    if (!currentUserId) {
+      throw new Error('User not found')
+    }
+
+    const chatId = `${audience}${contentType ? `:${contentType}` : ''}${
+      contentId ? `:${contentId}` : ''
+    }`
+
+    // Optimistically go to the chat. If we fail to create it, we'll toast
+    if (!skipNavigation) {
+      yield* put(goToChat({ chatId, presetMessage, replaceNavigation }))
+    }
+
+    // TODO: fetch chat history
+    // try {
+    //   yield* call(doFetchChatIfNecessary, { chatId })
+    // } catch {}
+    const existingChat = yield* select((state) => getChat(state, chatId))
+    if (!existingChat) {
+      const newBlast: ChatBlast = {
```

Comment:
nit: do we need the explicit type still?

Reply:
yeah, removing it we need a cast later